### DISCUSSION TITLE
What is the difference between the timeline focus and flamegraph?

### DISCUSSION THREAD

**Discussion by nickchomey** [NoiseByNorthwest/php-spx#274] [Q&A] [answered]
Thanks very much for this fantastic tool! However, one thing isn't clear to me at all.

Other than the timeline focus allowing for zooming etc, what is the difference between it and the flamegraph at the bottom? They seem to present fundamentally the same data, but the flamegraph is flipped and seems to have less detail. 

What is the purpose of having both? Does the flamegraph allows for different analysis/interpretation?

Thanks!

**Comment by nickchomey** [comment]
[This article](https://medium.com/performance-engineering-for-the-ordinary-barbie/profiling-flame-chart-vs-flame-graph-7b212ddf3a83) seems to provide some useful info 

The flamegraph at the bottom seems to aggregate the same function, and perhaps even sort alphabetically. So, it doesnt follow the precise timeline. 

I'd love for more insights though if anyone has them!

**Reply by NoiseByNorthwest** [reply, marked as answer]
I confirm that the linked article answers your question.
The timeline view (Flame chart in the article)  is easy to understand. The Flame graph is less obvious, it shows an aggregation per stack paths, this is why:
- one span can relate to many calls (of the same function called by a given parent stack path)
- the horizontal axis does not represent the timeline, but only the elapsed time by the span

In the end it allows to quickly spot the hot stack path if there is one.
